airflow/providers_manager.py

```diff
@@ -62,6 +64,7 @@ def __init__(self):
         self.__find_all_airflow_builtin_providers_from_local_sources()
         self.__find_all_providers_from_packages()
         self.__creates_provider_directory()
+        self.__retrieve_extra_links()
```

Comment:
Different verb used here (`find` vs `retrieve`) I think we should be consistent. Perhaps renaming all of them to `discover_*`?

Reply:
Discover is a good name.